aws/resource_aws_dlm_lifecycle_policy.go

```diff
@@ -0,0 +1,341 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/dlm"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsDlmLifecyclePolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDlmLifecyclePolicyCreate,
+		Read:   resourceAwsDlmLifecyclePolicyRead,
+		Update: resourceAwsDlmLifecyclePolicyUpdate,
+		Delete: resourceAwsDlmLifecyclePolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile("^[0-9A-Za-z _-]+$"), "see https://docs.aws.amazon.com/cli/latest/reference/dlm/create-lifecycle-policy.html"),
+				//	TODO: https://docs.aws.amazon.com/dlm/latest/APIReference/API_LifecyclePolicy.html#dlm-Type-LifecyclePolicy-Description says it has max length of 500 but doesn't mention the regex but SDK and CLI docs only mention the regex and not max length. Check this
+			},
+			"execution_role_arn": {
+				// TODO: Make this not required and if it's not provided then use the default service role, creating it if necessary
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+			"policy_details": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"resource_types": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"schedule": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"create_rule": {
+										Type:     schema.TypeList,
+										Required: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"interval": {
+													Type:     schema.TypeInt,
+													Required: true,
+													// TODO: Validate this is 12 or 24
```

Comment:
Is there a nice integer equivalent for `StringInSlice` validation function? Or do I need to write a small validation function for this? Couldn't see one anywhere but might have missed it.

Reply:
Not currently 😅  https://godoc.org/github.com/hashicorp/terraform/helper/validation

For now we can add one to `aws/validators.go` 👍 

```go
// IntInSlice returns a SchemaValidateFunc which tests if the provided value
// is of type int and matches the value of an element in the valid slice
func IntInSlice(validInts []int, ignoreCase bool) schema.SchemaValidateFunc {
	return func(i interface{}, k string) (s []string, es []error) {
		v, ok := i.(int)
		if !ok {
			es = append(es, fmt.Errorf("expected type of %s to be int", k))
			return
		}

		for _, validInt := range validInts {
			if v == validInt {
				return
			}
		}

		es = append(es, fmt.Errorf("expected %s to be one of %v, got %s", k, valid, v))
		return
	}
}
```

Reply:
Looks like @ewbankkit got around to adding this so I dropped my commit that added it after seeing it get pulled in on the rebase against master.